addons/udes_sale_stock/models/sale_order.py

```diff
@@ -267,6 +277,55 @@ def get_current_demand(self, products=None):
 
         return demand
 
+    def confirm_orders(self):
+        """Process sale and cancel, confirm or ignore sale orders according to
+           order processing configuration.
+        """
+        to_confirm = self.search([('state', '=', 'draft')])
+
+        for _, batch in to_confirm.batched(size=1000):
+            tries = 0
+            while True:
+                try:
+                    with self.env.cr.savepoint():
+                        batch.action_confirm()
+                        break
+                except OperationalError as e:
```

Comment:
why do a while loop as mat retries is a fixed value and we break on success anyway?

Reply:
Just to mimic Odoo's retry behaviour, same as we did for stock reservation.

Reply:
Fair enough